File: Java/p374.java

import java.util.Scanner;
import java.math.BigInteger;
import java.util.NoSuchElementException;

public class p374 {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        while (input.hasNext()) {
            BigInteger b = new BigInteger(input.next());
            BigInteger p = new BigInteger(input.next());
            BigInteger m = new BigInteger(input.next());

            b = b.modPow(p, m);

            System.out.println(b.toString());

            try{
                input.nextLine();
            } catch (NoSuchElementException e) {
                
            }
        }
    }
}

File: Python/p374.py
def BigMod(a, p, m):
    if p == 0: return 1
    if p % 2:
        return (a % m) * (BigMod(a, p - 1, m))
    else:
        c = BigMod(a, p / 2, m)
        return ((c % m) * (c % m)) % m


while True:
    try:
        B = int(input())
        P = int(input())
        M = int(input())
        s = input()
    except:
        break
    if P == 0:
        print(0)
    else:
        print(BigMod(B, P, M))
